plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -862,20 +869,52 @@ class FossId internal constructor(
                     }
                 }
 
-                emit(file to filteredSnippets.toSet())
+                val fileWithoutPrefix = archivePrefix?.let { file.removePrefix(it) } ?: file
+
+                emit(fileWithoutPrefix to filteredSnippets.toSet())
             }
         }
 
-        return RawResults(
+        return RawResults.createAndRemovePrefix(
             identifiedFiles,
             markedAsIdentifiedFiles,
             listIgnoredFiles,
             pendingFiles,
             snippets,
-            matchedLines
+            matchedLines,
+            archivePrefix
         )
     }
 
+    /**
+     * When the scan is created in Archive upload mode, all the file paths are prefixed with the archive name e.g.
+     * fossid-source-archive832364312005325574.zip/test.txt. This function searches this prefix in the list of pending
+     * files or in the list of identified files. If no such prefix is found, null is returned.
+     */
+    internal fun getArchivePrefix(pendingFiles: List<String>, identifiedFiles: List<IdentifiedFile>): String? {
+        var prefix = pendingFiles.firstOrNull {
+            it.startsWith("fossid-source-archive")
+        }?.substringBefore("/")
+
+        if (prefix == null) {
+            prefix = identifiedFiles.firstOrNull {
+                it.file.path?.startsWith("fossid-source-archive") ?: false
+            }?.file?.path?.substringBefore("/")
+        }
+
+        if (prefix == null) {
+            prefix = identifiedFiles.firstOrNull {
+                it.file.path?.startsWith("fossid-source-archive") ?: false
+            }?.file?.path?.substringBefore("/")
+        }
```

Comment:
Is this really required twice? (Compare to above.)

Reply:
We need this strange design because the prefix cannot be just removed when the results are mapped : it need to be reapplied later in the scanner when the requests for the snippet choice are made.